reactive/kotlinx-coroutines-reactive/src/Publish.kt

```diff
@@ -148,6 +153,12 @@ private class PublisherCoroutine<in T>(
             if (_nRequested.value >= CLOSED) {
                 _nRequested.value = SIGNALLED // we'll signal onError/onCompleted (that the final state -- no CAS needed)
                 val cause = getCompletionCause()
+                // Specification requires that after cancellation requested we don't call onXXX
+                if (cancelled) {
+                    // but we cannot just ignore exception so we handle it
+                    if (cause != null) handleCoroutineException(context, cause, this)
```

Comment:
Indeed. The whole code is wrong with respect to modern state of error-handling. It signals completion "onCancellation" which is simply a wrong way to do, since the final exception is not there yet. 

I've created a test that reproduces this problem (`PublishTest.testHandleFailureAfterCancel`) and fixed it by handling errors after completion.